src/api/aliases/getAliases.ts

```diff
@@ -40,14 +40,13 @@ export const getAliases = async (
   const jsonResponse = await rawResponse.json();
   const parsedRes = GetAliasesResponse.safeParse(jsonResponse);
 
-  if(rawResponse.status >= 200 && rawResponse.status < 300) {
-    if (parsedRes.success) {
-      return parsedRes.data;
-    }
+  // parsedRes and error {} from HOPRd have the same type,
+  // we can only rely on rawResponse.ok to know if its a success
+  if (parsedRes.success && rawResponse.ok) {
```

Comment:
What about:

```
if (rawResponse.ok) {
    const parsedRes = GetAliasesResponse.safeParse(jsonResponse);
    if (parsedRes.success) {
        return parsedRes.data;
    }
    throw new ZodError(parsedRes.error.issues);
}

const isApiErrorResponse = ApiErrorResponse.safeParse(jsonResponse);

  if (isApiErrorResponse.success) {
    throw new sdkApiError({
      status: rawResponse.status,
      statusText: rawResponse.statusText,
      hoprdErrorPayload: isApiErrorResponse.data
    });
  }

  // we could not parse the response and it is not unexpected
  throw new ZodError(isApiErrorResponse.error.issues);
```

Wouldn't that solve our issues?

Reply:
fixed